cg/meta/create_validation_cases/validation_cases_api.py

```diff
@@ -0,0 +1,185 @@
+"""Module that holds the api to create validation cases."""
+
+import logging
+import os
+from pathlib import Path
+
+from housekeeper.store.models import File, Version
+
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.constants import SequencingFileTag
+from cg.constants.housekeeper_tags import VALIDATION_TAG
+from cg.meta.create_validation_cases.validation_case_data import ValidationCaseData
+from cg.meta.create_validation_cases.validation_data_input import ValidationDataInput
+from cg.store.models import Case, CaseSample, Sample
+from cg.store.store import Store
+from cg.utils.files import copy_file, get_files_matching_pattern, rename_file
+
+LOG = logging.getLogger(__name__)
+
+
+class CreateValidationCaseAPI:
+
+    def __init__(self, status_db: Store, housekeeper_api: HousekeeperAPI, dry_run: bool = False):
+        self.status_db: Store = status_db
+        self.hk_api: HousekeeperAPI = housekeeper_api
+        self.dry_run = dry_run
+
+    def get_validation_case_data(self, validation_data_input: ValidationDataInput):
+        return ValidationCaseData(
+            status_db=self.status_db,
+            validation_data_input=validation_data_input,
+        )
+
+    def store_validation_samples(self, validation_case_data: ValidationCaseData) -> None:
+        """
+        Add a validation sample entry to StatusDB.
+        Raises:
+            ValueError
+        """
+        for sample in validation_case_data.validation_samples:
+            if self.status_db.sample_with_id_exists(sample_id=sample.internal_id):
+                raise ValueError(f"Sample {sample.internal_id} already exists in StatusDB.")
+            LOG.info(
+                f"New validation sample created: {sample.internal_id} from {sample.from_sample}"
+                f"Application tag set to: {sample.application_version.application.tag}"
+            )
+            if not self.dry_run:
+                self.status_db.session.add(sample)
+                LOG.info(f"Added {sample.name} to StatusDB.")
```

Comment:
It might make sense to move logic like this into the store where other pure data access logic is. In general, maybe we don't want to touch the session outside of the `Store` layer.